gen_mocks.go

```diff
@@ -114,6 +114,40 @@ func (v visitFn) Visit(node ast.Node) ast.Visitor {
 	}
 }
 
+func genFuncDecl(mockTypeName string, methName string, method *ast.FuncType) *ast.FuncDecl {
+	f := &ast.FuncDecl{
+		Recv: &ast.FieldList{List: []*ast.Field{
+			{
+				Names: []*ast.Ident{ast.NewIdent("s")},
+				Type:  &ast.StarExpr{X: ast.NewIdent(mockTypeName)},
+			},
+		}},
+		Name: ast.NewIdent(methName),
+		Type: method,
+		Body: &ast.BlockStmt{List: []ast.Stmt{
+			&ast.ExprStmt{
+				&ast.CallExpr{
+					Fun: &ast.SelectorExpr{
+						X:   ast.NewIdent("s"),
+						Sel: ast.NewIdent(methName + "_"),
+					},
+					Args:     fieldListToIdentList(method.Params),
+					Ellipsis: ellipsisIfNeeded(method.Params),
+				},
+			},
+		}},
+	}
+	if method.Results != nil {
+		f.Body = &ast.BlockStmt{List: []ast.Stmt{
+			&ast.ReturnStmt{
+				Results: []ast.Expr{f.Body.List[0].(*ast.ExprStmt).X},
```

Comment:
Referring to the just-set `f.Body` here obfuscates the logic. It would be clearer to create `f` and then have `f.Body` set in a subsequent `if method.Results != nil { ... } else { ... }` block.